docs/src/literate/quasi_incompressible_hyperelasticity.jl

```diff
@@ -0,0 +1,362 @@
+# [`incompressible_elasticity`](@__NBVIEWER_ROOT_URL__/examples/incompressible_elasticity.ipynb) and the incompressible analogue of
+# [`hyperelasticity`](@__NBVIEWER_ROOT_URL__/examples/hyperelasticity.ipynb). Much of the code therefore follows from the above two examples.
+# The problem is formulated in the undeformed or reference configuration with the displacement `u` and pressure `p` being the unknown fields. We now briefly outline
+# the formulation. Consider the standard hyperelasticity problem 
+#
+# ```math
+#   u = \argmin_{v\in\mathcal{K}}\Pi(v),\quad \text{where}\quad \Pi(v)  = \int_\Omega \Psi(v) d\Omega .
+# ```
+# For clarity of presentation we ignore any non-zero surface tractions and body forces and instead consider only
+# applied displacements (i.e. non-homogeneous dirichlet boundary conditions). Moreover we stick our attention to the standard Neo-Hookean stored energy density 
+# 
+# ```math
+#     \Psi(u) = \frac{\mu}{2}\left(I_1 - 3 \right) - \mu \log(J) + \frac{\lambda}{2}\left( J - 1\right){}^2.
+# ```
+# where I₁=F:F≡FᵢⱼFᵢⱼ and J = det(F) denote the standard invariants of the deformation gradient tensor F = 𝛪 + ∇u.
+# The above problem is ill-posed in the limit of incompressibility (or near-incompressibility), namely when
+# ```math
+#     \lambda/\mu \rightarrow +\infty
+# ```
+# In order to alleviate the problem, we consider the partial legendre transform of the strain energy density Ψ
+# ```math
+#   \widehat{\Psi}(p) = \sup_{J} p(J - 1) - \frac{\mu}{2}\left(I_1 - 3 \right) + \mu \log(J) - \frac{\lambda}{2}\left( J - 1\right){}^2
+# ```
+# where $J^\star$ is given by
+# ```math
+# J^\star = \arg\max \left[ p(J - 1) - \frac{\mu}{2}\left(I_1 - 3 \right) + \mu \log(J) - \frac{\lambda}{2}\left( J - 1\right){}^2\right]
+# ```
+# which yields
+# ```math
+#   J^\star(p) = \frac{\lambda + p + \sqrt{(\lambda + p)^2 + 4 * \lambda * \mu }}{(2 * \lambda}
+# ```
+# Taking the partial legendre transform of $\widehat{\Psi}$ gives us back the original problem now reformulated as
+# ```math
+#   \min_{u\in\mathcal{K}}\max_{p} \int_\Omega\Psi^{\star} (u, p)
+# ```
+# where $\Psi^\star (u, p)$ is given by
+# ```math
+#   \int_\Omega p (J - J^\star) d\Omega + \int_\Omega \frac{\mu}{2} \left( I_1 - 3\right) d\Omega - \int_\Omega \mu\log(J^\star)d\Omega + \int_\Omega \frac{\lambda}{2}\left( J^\star - 1 \right){}^2
+# ```
+# The euler-lagrange equations are then given by
+# ```math
+#   \int_\Omega \frac{\partial\Psi^\star}{\partial F}:\delta F d\Omega = 0 
+# ```
+# and
+# ```math
+#   \int_\Omega \frac{\partial \Psi^\star}{\partial p}\delta p d\Omega
+# ```
+# where δF = δ∇u = ∇(δu) and δu and δp denote test functions for the displacement and pressure fields respectively.
+# In order to apply Newton's method to the above problem, we need to calculate the respective hessians (tangent),
+# namely, ∂²Ψ*/∂F∂F, ∂²Ψ*/∂p² and ∂²Ψ*/∂F∂p which can be determined conveniently from automatic differentiation. 
```

Comment:
Slight inconsistency here with "∂F∂F" against "∂p²"

Reply:
Absolutely right. Inconsistent naming across functions and the text. I use the latter in Ψ and the former in text. I think I should keep the notation consistent. Thanks, will update this in a bit.

Reply:
I have a question Dennis: maybe this is more appropriate to ask on `Literate.jl` issues, but locally converting `<example>.jl` into `<example>.md` doesn't replace the ` ```math` with corresponding `$$`. Would you happen to know why? 

I can manually do the changes just to see if the example looks proper, but I guess that's the whole point of `Literate`, right?

Reply:
Ok, maybe we can continue it at another time. I will fiddle around with `Literate` in any case :)

Reply:
I think this is the correct behavior. AFAIK the "julian style" for math blocks in markdown is utilizing ``` over $$ . See e.g. here https://github.com/goessner/markdown-it-texmath .

Reply:
Nothing wrong with the markdown renderer. vs code simply does not support this style.